lru.c

```diff
@@ -409,9 +409,46 @@ LRU_setdefault(LRU *self, PyObject *args)
 
     if (lru_ass_sub(self, key, default_obj) != 0)
         return NULL;
+
+    Py_INCREF(default_obj);
     return default_obj;
 }
 
+static PyObject *
+LRU_pop(LRU *self, PyObject *args)
+{
+    PyObject *key;
+    PyObject *default_obj = NULL;
+    PyObject *result;
+
+    if (!PyArg_ParseTuple(args, "O|O", &key, &default_obj))
+        return NULL;
+
+    result = lru_subscript(self, key);
+
+    if (result) {
+        /* Save and restore Python error indicator around deleting item --
+         * because between the last call of item getter (lru_subscript) and
+         * next call of item setter (lru_ass_sub), the item may have been
+         * removed from the LRU as the effect of arbitrary action in code
+         * elsewhere, thus incurring an error that will not be cleared by us.
+         */
+
+        /* Question: does this matter in the current implementation? */
+        PyObject *e_type, *e_value, *e_traceback;
+
+        PyErr_Fetch(&e_type, &e_value, &e_traceback);
```

Comment:
Why is this required? Is it for handling updates via multiple threads?

Reply:
Good question, I'm not entirely sure if this is required, but I was playing safe, just in case that the call into the C-extension wasn't atomic from the point of view of the interpreter.

Reply:
I removed the unnecessary code here.